theano/configdefaults.py

```diff
@@ -1555,6 +1555,12 @@ def filter_vm_lazy(val):
              IntParam(5, lambda i: i > 0, allow_override=False),
              in_c_key=False)
 
+AddConfigVar('disbale_cycle_detection',
```

Comment:
I think I would use a different flags type, in case we had more option in the futur about about a flag: cycle_detection, that default to "topo" and the new version would be "fast". The next version of fast that allow some sequence of inputs could be given another name like sequence. (or replace fast if fast don't have advantage over it. I think it would be that case)